src/components/utils/test/posix_thread_test.cc

```diff
@@ -41,287 +43,466 @@ namespace utils_test {
+    thread->SetNameForId(thread->ThreadHandle(), kThreadName);
+    // Name must be large enough to keep 16 symbols. Read previous comment
+    char name[kMaxSize];
+    int result = pthread_getname_np(thread->ThreadHandle(), name, sizeof(name));
+    if (!result) {
+      EXPECT_EQ(kThreadName, std::string(name));
+    }
+    threadDelegate->WaitFor(test_lock, 10000);
+    // Stop thread
+    thread->Stop(threads::Thread::kThreadStopTask);
+    EXPECT_TRUE(
+        thread->Start(threads::ThreadOptions(threads::Thread::kMinStackSize)));
+    thread2_id = thread->CurrentId();
+    // Expect the same thread started with the the same name
+    EXPECT_EQ(kThreadName, std::string(name));
+    EXPECT_EQ(thread1_id, thread2_id);
+    threadDelegate->WaitFor(test_lock, 10000);
+  }
+
   EXPECT_TRUE(threadDelegate->check_value());
+
+  delete threadDelegate;
+  EXPECT_EQ(NULL, thread->GetDelegate());
   DeleteThread(thread);
+}
+
+TEST_F(PosixThreadTest, StartThreadWithNullPtrDelegate_ExpectThreadStateNone) {
+  threads::Thread* thread = nullptr;
+  // Create thread
+  ASSERT_NO_THROW(thread = CreateThread(kThreadName, nullptr));
+  EXPECT_TRUE(nullptr != thread);
+  EXPECT_EQ(nullptr, thread->GetDelegate());
+
+  EXPECT_FALSE(thread->Start());
+  EXPECT_FALSE(thread->IsRunning());
+  EXPECT_EQ(threads::Thread::kThreadStateNone, thread->thread_state_);
+
+  DeleteThread(thread);
+}
+
+TEST_F(PosixThreadTest,
+       StartThreadExecutingThreadMain_ExpectThreadStateRunning) {
+  threads::Thread* thread = nullptr;
+  const bool idle_threadMain = true;
+  TestThreadDelegate* threadDelegate = new TestThreadDelegate(idle_threadMain);
+  // Create thread
+  ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+  EXPECT_TRUE(nullptr != thread);
+
+  {
+    AutoLock auto_lock(*threadDelegate);
+    EXPECT_TRUE(thread->Start());
+    threadDelegate->WaitFor(auto_lock, 5000);
+  }
+
+  EXPECT_TRUE(thread->IsRunning());
+  EXPECT_EQ(threads::Thread::kThreadStateRunning, thread->thread_state_);
+
+  delete threadDelegate;
+  DeleteThread(thread);
+}
+
+TEST_F(PosixThreadTest, StartThreadExecutingThreadMain_ExpectThreadStateIdle) {
+  threads::Thread* thread = nullptr;
+  const bool idle_threadMain = true;
+  TestThreadDelegate* threadDelegate = new TestThreadDelegate(idle_threadMain);
+  // Create thread
+  ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+  EXPECT_TRUE(nullptr != thread);
+
+  {
+    AutoLock auto_lock(*threadDelegate);
+    EXPECT_TRUE(thread->Start());
+    threadDelegate->WaitFor(auto_lock, 5000);
+  }
+
+  thread->Stop(threads::Thread::kThreadStopTask);
+
+  EXPECT_FALSE(thread->IsRunning());
+  EXPECT_EQ(threads::Thread::kThreadStateIdle, thread->thread_state_);
+
   delete threadDelegate;
-  EXPECT_EQ(NULL, thread->delegate());
+  DeleteThread(thread);
+}
+
+TEST_F(PosixThreadTest,
+       StartThreadExecutingThreadMain_ExpectThreadStateCompleted) {
```

Comment:
add thread_command to test name

Reply:
@LuxoftAKutsan 
Sorry, :( please explain

Reply:
Multiple test names contain the same precondition but different effects in names. Please give names that contains different preconditions

Reply:
@LuxoftAKutsan 
Thanks, fixed in:https://github.com/LuxoftSDL/sdl_core/pull/50/commits/b04081c261b0fd3357a60daa185512deacfa12df